### PR TITLE
Resolve logically dead code from Coverity report

### PR DESCRIPTION
**Pull request by antonwolfy**
The PR propose to get rid of logically dead python code, which was reported by Coverity tool.
The PR is not intended to update any test or backend code, any work around that is proposed to be done by separate issue.

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: dpnp/dpnp_iface_mathematical.py [hunk 1]
```diff
@@ -442,58 +442,96 @@ def convolve(a, v, mode="full"):
...
+    >>> 1 / np.copysign(np.array(0), 1)
+    array(inf)
+    >>> 1 / np.copysign(np.array(0), -1)
+    array(-inf)
+
+    >>> x = np.array([-1, 0, 1])
+    >>> np.copysign(x, -1.1)
+    array([-1., -0., -1.])
+    >>> np.copysign(x, np.arange(3) - 1)
+    array([-1., -0., -1.])
```
Comment (vtavana) [inline dpnp/dpnp_iface_mathematical.py:481]:
```suggestion
    array([-1., 0., 1.])
```

Comment (antonwolfy) [reply dpnp/dpnp_iface_mathematical.py:481]: great catch, thank you!

File: dpnp/dpnp_iface_mathematical.py [hunk 2]
```diff
@@ -442,58 +442,96 @@ def convolve(a, v, mode="full"):
...
+        The values of `x1` with the sign of `x2`.
+
     Limitations
     -----------
-    Parameters `x1` and `x2` are supported as either :class:`dpnp.ndarray` or scalar.
-    Parameters `dtype`, `out` and `where` are supported with their default values.
+    Parameters `x1` and `x2` are supported as either scalar, :class:`dpnp.ndarray`
+    or :class:`dpctl.tensor.usm_ndarray`, but both `x1` and `x2` can not be scalars at the same time.
+    Parameters `where`, `dtype` and `subok` are supported with their default values.
     Keyword arguments `kwargs` are currently unsupported.
```
[the remark this answers is not in the document]
Comment (antonwolfy) [reply dpnp/dpnp_iface_mathematical.py:463]: updated for all the files

**Review by vtavana — APPROVED** [review]
Thank you, @antonwolfy

**Pull request merged**